drivers/hmis/app/graphql/types/hmis_schema/ce_referral_step.rb

```diff
@@ -8,37 +8,72 @@
+    #   step: Hmis::WorkflowExecution::Step,
+    #   referral: Hmis::Ce::Referral,
+    # }
+    # Why: The Step schema object should resolve this step's potential Participants (ReferralParticipants on the Referral).
+    # However, the Step model does not have a direct relationship to Referral, to avoid a duplicative bidirectional relationship.
 
     field :id, ID, null: false, description: 'unique identifier for this step based on node and instance'
     field :step_id, ID, null: true, method: :id, description: 'the DB identifier of this step, if it is persisted'
     field :form_definition, Types::Forms::FormDefinition, null: false
     field :name, String, null: false
     field :status, HmisSchema::Enums::CeReferralStepStatus, null: false
-    field :swimlane, String, null: false
     field :submitted_values, JsonObject, null: true
-    delegate :name, to: :workflow_node
+    field :swimlane, HmisSchema::CeReferralSwimlane, null: true, description: 'Swimlane for this step, which holds information about potential step participants'
+    field :assignees, [Application::User], null: true, description: 'User(s) currently assigned to this step'
 
     def id
       # the step may not yet be persisted, such as when it isn't yet available in the workflow
-      "#{object.node_id}:#{object.instance_id}"
+      "#{object.step.node_id}:#{object.step.instance_id}"
+    end
+
+    def step_id
+      object.step.id
+    end
+
+    def name
+      load_ar_association(object.step, :node).name
+    end
+
+    def status
+      object.step.status
+    end
+
+    def submitted_values
+      object.step.submitted_values
     end
 
     def workflow_node
-      load_ar_association(object, :node)
+      load_ar_association(object.step, :node)
     end
 
     def swimlane
-      load_ar_association(object, :swimlane)&.name
+      swimlane = load_ar_association(object.step, :swimlane)
+      participants_by_swimlane_id = object.referral.participants.group_by(&:swimlane_id)
+
+      return nil unless swimlane.present?
+
+      OpenStruct.new(
+        id: swimlane.id,
+        name: swimlane.name,
+        participants: participants_by_swimlane_id[swimlane.id]&.map(&:user) || [],
+      )
+    end
+
+    def assignees
+      load_ar_association(object.step, :assignments)&.map(&:user)
     end
 
     def form_definition
       # If the step has been submitted before, it stores a reference to the definition it was submitted with
-      definition = load_ar_association(object, :form_definition)
+      definition = load_ar_association(object.step, :form_definition)
       return definition if definition.present?
 
       # Otherwise, get the definition identifier on the node, and return the latest published definition with this identifier
-      node = load_ar_association(object, :node)
+      node = load_ar_association(object.step, :node)
       load_ar_association(node, :form_definitions, scope: Hmis::Form::Definition.published.order(version: :desc)).first
```

Comment:
since we're removing `scope:` maybe we can do that here. if you take the other suggestion about adding a helper for loading the task, this might be:
```ruby
load_ar_association(workflow_task, :form_definitions).select(&:published?).first
```

Reply:
Thanks! On my draft of that work, I'm proposing to remove `load_ar_association`s from this method since it's not resolved in batch. https://github.com/greenriver/hmis-warehouse/pull/5314/files?diff=split&w=0#diff-9a2e4184fa85c9a6f6a65383e8986f26ad77ff3c51716c6cb6a66cced4421b4d

Reply:
great!